src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -288,7 +307,11 @@ private void encodeChainedFeatures(String attributeName, List<Feature> chainedFe
             }
         }
         // end the JSON chained features array
-        jsonWriter.endArray();
+        if (!isInsideArrayAttributeName(attributeName)) jsonWriter.endArray();
+    }
+
+    private boolean isInsideArrayAttributeName(String attributeName) {
```

Comment:
No idea what this is, could you add documentation?

Reply:
Added